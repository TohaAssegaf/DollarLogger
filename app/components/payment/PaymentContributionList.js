import styles from './styles'
import PaymentContributionCell from './PaymentContributionCell'
import React from 'react'
import { FlatList } from 'react-native'

export default class PaymentContributionList extends React.Component {
  keyExtractor(paymentContribution, index) {
    return String(index)
  }

  renderItem(paymentContribution) {
    return <PaymentContributionCell
      onTap={() => this.props.onTapCell(paymentContribution)}
      paymentContribution={paymentContribution}
    />
  }

  render() {
    return (
      <FlatList
        data={this.props.paymentContributions}
        keyExtractor={(paymentContribution, index) => this.keyExtractor(paymentContribution, index)}
        renderItem={({item}) => this.renderItem(item)}
        style={styles.paymentContributionList}
      />
    );
  }
}
