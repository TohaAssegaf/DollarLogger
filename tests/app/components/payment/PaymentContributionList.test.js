import PaymentContributionList from '~/app/components/payment/PaymentContributionList'
import React from 'react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const rendered = renderer.create(<PaymentContributionList paymentContributions={[
    {
      displayName: "Test payment (1/2)",
      total: 100,
      date: new Date(2018, 4, 2),
      id: 1
    },
    {
      displayName: "Test payment (2/2)",
      total: 100,
      date: new Date(2018, 4, 9),
      id: 1
    },
  ]}
  />).toJSON();
  expect(rendered).toMatchSnapshot();
});
