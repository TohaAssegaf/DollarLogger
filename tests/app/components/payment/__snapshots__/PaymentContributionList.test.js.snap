// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "date": 2018-05-02T04:00:00.000Z,
        "displayName": "Test payment (1/2)",
        "id": 1,
        "total": 100,
      },
      Object {
        "date": 2018-05-09T04:00:00.000Z,
        "displayName": "Test payment (2/2)",
        "id": 1,
        "total": 100,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "#222222",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 60,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.15,
                "paddingHorizontal": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              5/2
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 0.6,
              }
            }
          >
            <Text
              numberOfLines={2}
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              Test payment (1/2)
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.25,
                "paddingRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              $1.00
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "#222222",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 60,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.15,
                "paddingHorizontal": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              5/9
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 0.6,
              }
            }
          >
            <Text
              numberOfLines={2}
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              Test payment (2/2)
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.25,
                "paddingRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              $1.00
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
